grpc-server/hardware.go

```diff
@@ -23,39 +23,33 @@ func (s *server) Push(ctx context.Context, in *hardware.PushRequest) (*hardware.
 	// must be a copy so deferred cacheInFlight.Dec matches the Inc
 	labels = prometheus.Labels{"method": "Push", "op": ""}
 
-	var h struct {
-		ID    string
-		State string
-	}
-	err := json.Unmarshal([]byte(in.Data), &h)
-	if err != nil {
+	hw := in.Data
+	if hw.Id == "" {
 		metrics.CacheTotals.With(labels).Inc()
 		metrics.CacheErrors.With(labels).Inc()
-		err = errors.Wrap(err, "unmarshal json")
+		err := errors.New("id must be set to a UUID")
```

Comment:
We can indicate that current hw.Id is nil string in this case (add hw.Id to error log). This will give extra info for debugging.